pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go

```diff
@@ -327,3 +332,44 @@ func TestUnhealthyForTooLong(t *testing.T) {
 		}
 	}
 }
+
```

Comment:
This should be called `TestRemediation` and baremetal should be test case. That would also force you to have the baremetal domain logic well scoped in its own function even it that produces some duplication

Reply:
I think that BareMetal is the only provider that has different logic on how to remediate machines, so for me, it makes sense to have it under the different test case, also I hope in 4.3 we will have the same logic for all providers, so it is only short term solution